src/kfactory/kcell.py

```diff
@@ -11,6 +11,7 @@
 from __future__ import annotations
 
 import functools
+import hashlib
```

Comment:
We are already importing hashlib further down once. Albeit differently, but I also don't know what we would use hashlib otherwise for except shas.